File: docs-examples/example-java/src/test/java/io/pollend/quartz/docs/quickstart/MessageJobSpec.java
package io.pollend.quartz.docs.quickstart;

import io.micronaut.context.ApplicationContext;
import org.junit.jupiter.api.Test;
import org.quartz.JobBuilder;
import org.quartz.JobDataMap;
import org.quartz.Scheduler;
import org.quartz.SchedulerException;
import org.quartz.TriggerBuilder;

import java.util.HashMap;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;

public class MessageJobSpec {
    @Test
    void testJobCreation() throws SchedulerException {
        ApplicationContext ctx = ApplicationContext.run(new HashMap<String, Object>() {
            {
                put("quartz.enabled", true);
                put("quartz.clients.default.config-file", "classpath:default.properties");
            }
        });

// tag::create_job[]
Scheduler scheduler = ctx.getBean(Scheduler.class);
scheduler.scheduleJob(JobBuilder.newJob(MessageJob.class)
    .setJobData(new JobDataMap(new HashMap<String, Object>() {{
        put("message", "My Message");
    }})).build(), TriggerBuilder.newTrigger().startNow().build());
// end::create_job[]

        MessageJob job = ctx.getBean(MessageJob.class);
        try {
            await().atMost(5, SECONDS).until(() ->
                job.messages.size() == 1 &&
                    job.messages.get(0).equals("My Message"));
        } finally {
            ctx.close();
        }

    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/quartz/docs/quickstart/MessageJobSpec.kt
package io.micronaut.quartz.docs.quickstart

import io.kotlintest.eventually
import io.kotlintest.seconds
import io.kotlintest.shouldBe
import io.kotlintest.specs.BehaviorSpec
import io.micronaut.context.ApplicationContext
import org.opentest4j.AssertionFailedError
import org.quartz.JobBuilder
import org.quartz.JobDataMap
import org.quartz.Scheduler
import org.quartz.TriggerBuilder

class MessageJobSpec: BehaviorSpec() {
    init {
        given("test job creation") {
            val ctx = ApplicationContext.run(
                mutableMapOf(
                    "quartz.enabled" to true,
                    "quartz.clients.default.config-file" to "classpath:default.properties"
                )
            )
            `when`("the job is created") {
//                val job = ctx.getBean(MessageJob::class.java)

// tag::create_job[]
val scheduler = ctx.getBean(Scheduler::class.java)
scheduler.scheduleJob(JobBuilder.newJob(MessageJob::class.java).setJobData(
    JobDataMap(mutableMapOf(
        "message" to "My Message"
    ))
).build(), TriggerBuilder.newTrigger().startNow().build())
// end::create_job[]

                then("the message is retrieved") {
                    val job = ctx.getBean(MessageJob::class.java)
                    eventually(10.seconds, AssertionFailedError::class.java) {
                        job.messages.size shouldBe  1
                        job.messages.first() shouldBe "My Message"
                    }
                }
            }
        }
    }
}
